### PR TITLE
ZIO Test: Add Documentation to Mock Package

### PR DESCRIPTION
**Pull request by adamgfraser**

### CODE REVIEW

File: test/shared/src/main/scala/zio/test/mock/Live.scala [hunk 1]
```diff
@@ -28,16 +48,33 @@ object Live {
     def provide[E, A](zio: ZIO[R, E, A]): IO[E, A]
   }
 
+  /**
+   * Provide an effect with the "live" environment.
```
Comment (regiskuckaertz) [inline test/shared/src/main/scala/zio/test/mock/Live.scala]:
```suggestion
   * Provides an effect with the "live" environment.
```

File: test/shared/src/main/scala/zio/test/mock/Live.scala [hunk 2]
```diff
@@ -28,16 +48,33 @@ object Live {
...
+   */
   def live[R, E, A](zio: ZIO[R, E, A]): ZIO[Live[R], E, A] =
     ZIO.accessM[Live[R]](_.live.provide(zio))
 
+  /**
+   * Constructs a new `Live` instance with an environment `R`. This should
+   * typically not be necessary as `MockEnvironment` provides access to live
+   * versions of all the standard ZIO environmental effecs but could be useful
```
Comment (regiskuckaertz) [inline test/shared/src/main/scala/zio/test/mock/Live.scala]:
```suggestion
   * versions of all the standard ZIO environmental effects but could be useful
```

File: test/shared/src/main/scala/zio/test/mock/MockConsole.scala [hunk 3]
```diff
@@ -37,16 +71,27 @@ object MockConsole {
...
+    /**
+     * Writes the specified string to the output buffer.
+     */
     override def putStr(line: String): UIO[Unit] =
       consoleState.update { data =>
         Data(data.input, data.output :+ line)
       }.unit
 
+    /**
+     * Writes the specified string to the output buffer followed by a new line
```
Comment (regiskuckaertz) [inline test/shared/src/main/scala/zio/test/mock/MockConsole.scala]:
```suggestion
     * Writes the specified string to the output buffer followed by a newline
```

File: test/shared/src/main/scala/zio/test/mock/MockConsole.scala [hunk 4]
```diff
@@ -60,42 +105,91 @@ object MockConsole {
...
 
+  /**
+   * Accesses a `MockConsole` instance in the environment and writes the
+   * specified sequence of strings to the input buffer.
+   */
   def feedLines(lines: String*): ZIO[MockConsole, Nothing, Unit] =
     ZIO.accessM(_.console.feedLines(lines: _*))
 
+  /**
+   * Access a `MockConsole` instance in the environment and returns the
```
Comment (regiskuckaertz) [inline test/shared/src/main/scala/zio/test/mock/MockConsole.scala]:
```suggestion
   * Accesses a `MockConsole` instance in the environment and returns the
```

File: test/shared/src/main/scala/zio/test/mock/MockConsole.scala [hunk 5]
```diff
@@ -60,42 +105,91 @@ object MockConsole {
...
+   * contents of the output buffer.
+   */
   val output: ZIO[MockConsole, Nothing, Vector[String]] =
     ZIO.accessM(_.console.output)
 
+  /**
+   * Access a `MockConsole` instance in the environment and clears the input
```
Comment (regiskuckaertz) [inline test/shared/src/main/scala/zio/test/mock/MockConsole.scala]:
```suggestion
   * Accesses a `MockConsole` instance in the environment and clears the input
```

File: test/shared/src/main/scala/zio/test/mock/MockConsole.scala [hunk 6]
```diff
@@ -60,42 +105,91 @@ object MockConsole {
...
+   * buffer.
+   */
   val clearInput: ZIO[MockConsole, Nothing, Unit] =
     ZIO.accessM(_.console.clearInput)
 
+  /**
+   * Access a `MockConsole` instance in the environment and clears the output
```
Comment (regiskuckaertz) [inline test/shared/src/main/scala/zio/test/mock/MockConsole.scala]:
```suggestion
   * Accesses a `MockConsole` instance in the environment and clears the output
```

File: test/shared/src/main/scala/zio/test/mock/MockSystem.scala [hunk 7]
```diff
@@ -35,58 +53,122 @@ object MockSystem {
...
+  /**
+   * Accesses a `MockSystem` instance in the environment and clears the mapping
+   * of system properties.
+   */
   def clearProperty(prop: String): ZIO[MockSystem, Nothing, Unit] =
     ZIO.accessM(_.system.clearProperty(prop))
 
+  /**
+   * The default initial state of the `MockSystem` with no environment variable
+   * or system property mappings and the system line separate set to the new
```
Comment (regiskuckaertz) [inline test/shared/src/main/scala/zio/test/mock/MockSystem.scala]:
```suggestion
   * or system property mappings and the system line separator set to the new
```

File: test/shared/src/main/scala/zio/test/mock/package.scala [hunk 8]
```diff
@@ -20,6 +20,57 @@ import zio.{ IO, ZIO }
 
 import zio.Managed
 
+/**
+ * The `mock` package contains testable versions of all the standard ZIO
+ * environment types through the `MockClock`, `MockConsole`, `MockSystem`, and
```
Comment (regiskuckaertz) [inline test/shared/src/main/scala/zio/test/mock/package.scala]:
```suggestion
 * environment types through the [[MockClock]], [[MockConsole]], [[MockSystem]], and
```

File: test/shared/src/main/scala/zio/test/mock/package.scala [hunk 9]
```diff
@@ -20,6 +20,57 @@ import zio.{ IO, ZIO }
...
+ * `MockRandom` modules. See the documentation on the individual modules for
```
Comment (regiskuckaertz) [inline test/shared/src/main/scala/zio/test/mock/package.scala]:
```suggestion
 * [[MockRandom]] modules. See the documentation on the individual modules for
```

**Comment by jdegoes** [comment]
@adamgfraser Great to see such high quality documentation! ❤️

**Comment by adamgfraser** [comment]
@regiskuckaertz Thanks for taking the time to go through this! I incorporated all your changes.

**Pull request merged**